### PR TITLE
Overload `save` and `restore`

### LINKED ISSUE #207: Overload `save`/`restore` to fix serialisation when atomic model has ephemeral fitresult
https://github.com/alan-turing-institute/MLJ.jl/issues/1099

### PR DESCRIPTION
**Pull request by ablaom**
Resolves #207

### CODE REVIEW

**Comment by ablaom** [comment]
@paulotrefosco Be great if you can confirm this MLJTuning.jl branch resolves your issue with wrapping XGBoost models in TunedModel. 

You can add this branch to your environment with `using Pkg; Pkg.add(name="MLJTuning", rev="save-restore")`

File: test/tuned_models.jl [hunk 1]
```diff
@@ -406,4 +406,53 @@ end
...
+thing = []
+struct EphemeralRegressor <: Deterministic end
+function MLJBase.fit(::EphemeralRegressor, verbosity, X, y)
+    # if I serialize/deserialized `thing` then `view` below changes:
+    view = objectid(thing)
+    fitresult = (thing, view, mean(y))
+    return fitresult, nothing, NamedTuple()
+end
+function MLJBase.predict(::EphemeralRegressor, fitresult, X)
+    thing, view, μ = fitresult
```
Comment (OkonSamuel) [inline test/tuned_models.jl:420]: I think we should rename the `view` variable name.

**Review by OkonSamuel — CHANGES_REQUESTED** [review]
Looks good to me, except for the minor change

File: test/tuned_models.jl (hunk 1 above)
Comment (ablaom) [reply test/tuned_models.jl]: done

**Pull request merged**